vendor/github.com/coreos/go-oidc/oidc/client.go

```diff
@@ -621,6 +621,20 @@ func (c *Client) Healthy() error {
 
 	return nil
 }
+func (c *Client) OAuthClientWithClientSecretPost() (*oauth2.Client, error) {
```

Comment:
I don't think we need to make changes to the library to support the different `AuthMethod` I think there was support to make this configurable when you create the client?

Reply:
@russjones  I did review the flow and the `client_secret_basic` or `client_secret_post` auth method is set in the vendor code. The  original method `chooseAuthMethod` gets the auth methods which are pulled from the provider configuration from its server from the vendor code.  We can't change that setting outside the vendor code from what I can see.

Reply:
I think we'll need to tweak the code, I don't see a way to reliably manipulate `chooseAuthMethod()` and it's not otherwise configurable.

It looks like we've already forked go-oidc, so I assume we could make a PR to https://github.com/gravitational/go-oidc?

Reply:
I modified have a new SetAuthMethod in oauth2 that allows setting the auth method.